src/main/java/com/direwolf20/mininggadgets/common/items/MiningGadget.java

```diff
@@ -88,7 +88,10 @@ public int getMaxDamage(ItemStack stack) {
 
     @Override
     public boolean isBarVisible(ItemStack stack) {
-        IEnergyStorage energy = stack.getCapability(CapabilityEnergy.ENERGY, null).orElse(null);
+        if (UpgradeTools.containsActiveUpgradeFromList(UpgradeTools.getUpgrades(stack), Upgrade.BATTERY_CREATIVE))
```

Comment:
IIRC this is a relatively heavy method and this method is run every tick. I'd recommend evaluating the performance of this call.